confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/statemachine/RegionMigrateProcedure.java

```diff
@@ -87,6 +87,7 @@ protected Flow executeFromState(ConfigNodeProcedureEnv env, RegionTransitionStat
     try {
       switch (state) {
         case REGION_MIGRATE_PREPARE:
+          handler.markRegionAsMigrating(consensusGroupId, originalDataNode);
```

Comment:
This is a effective but not safe way to update RegionStatus. The migrating status will be lost if the ConfigNodeGroup switch the ConfigNode-leader during Region migration process. A better way to satisfy this feature is sending a RPC request to record this Removing status on DataNode. Therefore this Removing status will be sent to ConfigNode-leader in Cluster heartbeat. You can reference how ConfigNode currently identifies a Removing DataNode.